### PR TITLE
Sc 197028/robustify sampling configs

### PR DESCRIPTION
**Pull request by russelldeguzman**
1. Ensuring that interval == duration  == controller disabled..
2. Cool way of cbor map copy. 
3. bonus bug - interval being set to duration

### CODE REVIEW

File: src/apps/bridge/bridgePowerController.cpp [hunk 1]
```diff
@@ -192,11 +200,11 @@ void BridgePowerController::_update(
             BRIDGE_LOG_PRINT("Bridge State Subsampling Off\n");
             uint32_t nextSubsampleEpochS = _subSampleIntervalStartS + _subsampleIntervalS;
             _subSampleIntervalStartS = nextSubsampleEpochS;
-            time_to_sleep_ms =
-                MAX((nextSubsampleEpochS - currentCycleS) * 1000,
-                    MIN_TASK_SLEEP_MS);
-            if (_subSampleIntervalStartS !=
-                currentCycleS) { // Prevent bus thrash
+            time_to_sleep_ms = (currentCycleS < nextSubsampleEpochS) ?
```
Comment (russelldeguzman) [inline src/apps/bridge/bridgePowerController.cpp:203]: Handle sign flip

File: src/apps/bridge/bridgePowerController.cpp [hunk 2]
```diff
@@ -192,11 +200,11 @@ void BridgePowerController::_update(
...
-            time_to_sleep_ms =
-                MAX((nextSubsampleEpochS - currentCycleS) * 1000,
-                    MIN_TASK_SLEEP_MS);
-            if (_subSampleIntervalStartS !=
-                currentCycleS) { // Prevent bus thrash
+            time_to_sleep_ms = (currentCycleS < nextSubsampleEpochS) ? 
+              MAX((nextSubsampleEpochS - currentCycleS) * 1000, MIN_TASK_SLEEP_MS) :
+              MIN_TASK_SLEEP_MS;
+            // Prevent bus thrash
+            if (nextSubsampleEpochS > currentCycleS) {
```
Comment (russelldeguzman) [inline src/apps/bridge/bridgePowerController.cpp:207]: More intuitive name change

File: src/lib/common/topology_sampler.cpp [hunk 3]
```diff
@@ -531,156 +530,13 @@ static bool encode_cbor_configuration(CborEncoder &array_encoder,
...
-      break;
-    }
-    // Create a map in the subarray to encode to.
-    err = cbor_encoder_create_map(&array_encoder, &map_encoder, num_fields);
-    if(err != CborNoError) {
-      break;
+    memcpy(array_encoder.data.ptr, cbor_map_reply.cbor_data, cbor_map_reply.cbor_encoded_map_len);
+    array_encoder.data.ptr += cbor_map_reply.cbor_encoded_map_len;
+    if (array_encoder.remaining) {
+      array_encoder.remaining--;
```
Comment (russelldeguzman) [inline src/lib/common/topology_sampler.cpp:536]: Decrements the number of elements in the array expected.

**Review by towynlin — APPROVED** [review]
Woot! 🐞🐛🐜
And I love the massive deletion of that cbor decode-encode!

**Pull request merged**